src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -1604,6 +1637,43 @@ public ProxyPreferences getProxyPreferences() {
         return proxyPreferences;
     }
 
+    @Override
+    public SSLPreferences getSSLPreferences() {
+        if (Objects.nonNull(sslPreferences)) {
+            return sslPreferences;
+        }
+
+        sslPreferences = new SSLPreferences(
+                getBoolean(CUSTOM_CERTIFICATE_USE),
+                getStringList(CUSTOM_CERTIFICATE_THUMBPRINT),
+                getStringList(CUSTOM_CERTIFICATE_VERSION),
+                getStringList(CUSTOM_CERTIFICATE_VALID_FROM),
+                getStringList(CUSTOM_CERTIFICATE_VALID_TO),
+                getStringList(CUSTOM_CERTIFICATE_SERIAL_NUMBER),
+                getStringList(CUSTOM_CERTIFICATE_ISSUER),
+                getStringList(CUSTOM_CERTIFICATE_SIG_ALGORITHM),
+                get(TRUSTSTORE_PATH)
+        );
```

Comment:
Im guessing you read them as string lists to be able to observe them. There is maybe a solution to parse the certificates already here and to create an array of observables for every certificate. See BibEntry::getObservables . Would that be a cleaner solution?

Reply:
Speaking about it made me realize that we don't need to store that information anywhere else; it's already in the truststore. The disadvantage is that we can't store custom properties; we're stuck with what the `X509Certificate` class provides. I updated the code to use the truststore as the single source of truth, I think this is better, the alternative brings lots of syncing issues